tests/components/rfxtrx/test_sensor.py

```diff
@@ -226,83 +212,93 @@ async def test_update_of_sensors(hass, rfxtrx):
-            } == _entity_temp.device_state_attributes
-            assert "Bath" == _entity_temp.__str__()
-        elif id == "sensor_05_02":
-            device_num = device_num + 1
-            entity = rfxtrx_core.RFX_DEVICES[id]["Temperature"]
-            assert TEMP_CELSIUS == entity.unit_of_measurement
-            assert 13.3 == entity.state
-            assert {
-                "Humidity status": "normal",
-                "Temperature": 13.3,
-                "Rssi numeric": 6,
-                "Humidity": 34,
-                "Battery numeric": 9,
-                "Humidity status numeric": 2,
-            } == entity.device_state_attributes
-            assert "Test" == entity.__str__()
-
-    assert 2 == device_num
-    assert 2 == len(rfxtrx_core.RFX_DEVICES)
+    state = hass.states.get("sensor.test_temperature")
+    assert state
+    assert state.state == "unknown"
+    assert (
+        state.attributes.items()
+        >= {
+            "friendly_name": "Test Temperature",
+            "unit_of_measurement": TEMP_CELSIUS,
+        }.items()
+    )
+
+    state = hass.states.get("sensor.bath_temperature")
+    assert state
+    assert state.state == "unknown"
+    assert (
+        state.attributes.items()
+        >= {
+            "friendly_name": "Bath Temperature",
+            "unit_of_measurement": TEMP_CELSIUS,
+        }.items()
+    )
+
+    state = hass.states.get("sensor.bath_humidity")
+    assert state
+    assert state.state == "unknown"
+    assert (
+        state.attributes.items()
+        >= {
+            "friendly_name": "Bath Humidity",
+            "unit_of_measurement": UNIT_PERCENTAGE,
+        }.items()
+    )
+
+    assert len(hass.states.async_all()) == 3
+
+    await _signal_event(hass, "0a520802060101ff0f0269")
+    await _signal_event(hass, "0a52080705020085220269")
+
+    state = hass.states.get("sensor.test_temperature")
+    assert state
+    assert state.state == "13.3"
+    assert (
+        state.attributes.items()
+        >= {
+            "friendly_name": "Test Temperature",
+            "unit_of_measurement": TEMP_CELSIUS,
+            "Battery numeric": 9,
+            "Temperature": 13.3,
+            "Humidity": 34,
+            "Humidity status": "normal",
+            "Humidity status numeric": 2,
+            "Rssi numeric": 6,
+        }.items()
+    )
+
+    state = hass.states.get("sensor.bath_temperature")
+    assert state
+    assert state.state == "51.1"
+    assert (
+        state.attributes.items()
+        >= {
+            "friendly_name": "Bath Temperature",
+            "unit_of_measurement": TEMP_CELSIUS,
+            "Battery numeric": 9,
+            "Temperature": 51.1,
+            "Humidity": 15,
+            "Humidity status": "normal",
+            "Humidity status numeric": 2,
+            "Rssi numeric": 6,
+        }.items()
+    )
+
+    state = hass.states.get("sensor.bath_humidity")
+    assert state
+    assert state.state == "15"
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5802827)